src/assets/docs/ScMachineBuildAndContributionGuide.md

```diff
@@ -0,0 +1,387 @@
+
+## Установка проекта
+
+Установите sc-machine через git:
+
+```sh
+git clone https://github.com/ostis-ai/sc-machine
+cd sc-machine
+git submodule update --init --recursive
+```
+
+## Проверка CMake
+
+Сначала установите pipx, используя [**руководство по установке pipx**](https://pipx.pypa.io/stable/installation/), если он еще не установлен.
+
+Убедитесь, что используете **CMake версии 3.24** или новее. Проверьте вашу версию командой:
+
+```sh
+cmake --version
+```
+
+Для обновления CMake выполните:
+
+```sh
+# Use pipx to install cmake if not already installed
+pipx install cmake
+pipx ensurepath
+# relaunch your shell after installation
+exec $SHELL
+```
+
+Установите генератор Ninja для CMake, чтобы использовать пресеты sc-machine:
+
+```sh
+# Use pipx to install ninja if not already installed
+pipx install ninja
+pipx ensurepath
+# relaunch your shell after installation
+exec $SHELL
+```
+
+## Начало разработки sc-machine с использованием Conan
+
+### Установка Conan
+
+Установите Conan для сборки sc-machine с зависимостями, предоставляемыми Conan:
+
+```sh
+# Use pipx to install conan if not already installed
+pipx install conan
+pipx ensurepath
+# relaunch your shell after installation
+exec $SHELL
+```
+
+### Использование sc-machine в режиме Debug
+
+#### Сборка sc-machine в режиме Debug
+
+Для сборки sc-machine в режиме отладки с зависимостями от Conan выполните:
+
+```sh
+# debug build type
+cmake --preset debug-conan
+cmake --build --preset debug
+```
+
+> По умолчанию пресет `debug` включает сборку тестов sc-machine.
+
+#### Запуск тестов sc-machine в режиме Debug
+
+После этого перейдите в `build/Debug` и запустите тесты через `ctest`:
+
+```sh
+cd build/Debug
+ctest -V
+```
+
+#### Запуск sc-machine в режиме Debug
+
+```sh
+# create empty knowledge base sources folder
+mkdir kb
+# note: at this stage you can move your KB sources to the ./kb folder
+
+# build knowledge base
+./build/Debug/bin/sc-builder -i kb -o kb.bin --clear
+# run sc-machine
+./build/Debug/bin/sc-machine -e build/Debug/lib/extensions -s kb.bin
+```
+
+### Использование sc-machine в режиме Release
+
+#### Сборка sc-machine в режиме Release
+
+Для сборки sc-machine в режиме релиза с зависимостями от Conan выполните:
+
+```sh
+# release build type without tests
+cmake --preset release-conan
+cmake --build --preset release
+```
+
+Для сборки sc-machine с тестами в режиме релиза с зависимостями от Conan выполните:
+
+```sh
+# release build type with tests
+cmake --preset release-with-tests-conan
+cmake --build --preset release
+```
+
+#### Запуск тестов sc-machine в режиме Release
+
+После этого вы можете запустить тесты:
+
+```sh
+cd build/Release
+ctest -V
+```
+
+#### Запуск sc-machine в режиме Release
+
+```sh
+# создайте пустую папку с исходниками базы знаний
+mkdir kb
+# на этом этапе вы можете переместить ваши исходники KB в папку ./kb
+
+# сборка базы знаний
+./build/Release/bin/sc-builder -i kb -o kb.bin --clear
+# запуск sc-machine
```

Comment:
not all comments were translated

Reply:
text contains "sc-machine" in some places too. i think we can leave it as is, because it's a name